### PR TITLE
 [TIC 80] Correct active/inactive events meaning and allow for soft deletion

### PR DESCRIPTION
**Pull request by benfaustin95**
### Description

- Added soft delete service for both events and event instances. 
- Updated routes for events and event instances with Prisma.
- Updated API 1 routes so they do not return deleted items. 

### Risks
Missed updating a route(s) and deleted items are being sent to the frontend. 

### Validation
Testing of functionality through the web app as well as through Swagger. 

### Issue
NA

### Operating System
Apple Silicon

### CODE REVIEW

**Review by HaliTran — DISMISSED** [review]
Looks good to me! Thanks for updating the API's it's gonna help a lot moving forward!

File: server/src/controllers/eventController.ts [hunk 1]
```diff
@@ -9,11 +9,53 @@ import {
...
+ * /2/events/checkout:
+ *   post:
+ *     summary: Update contact information, create order, create Stripe checkout session
+ *     tags:
+ *     - New Event API
+ *     requestBody:
+ *       description: Checkout information
+ *       content:
+ *         application/json:
+ *           schema:
```
Comment (henrykaus) [inline server/src/controllers/eventController.ts:29]: Empty schema here

File: server/src/controllers/eventController.ts [hunk 2]
```diff
@@ -84,168 +126,100 @@ eventController.post('/checkout', async (req: Request, res: Response) => {
...
   }
 });
 
-eventController.use(checkJwt);
-eventController.use(checkScopes);
-
 /**
  * @swagger
- * /2/event:
+ * /2/events/:
```
Comment (henrykaus) [inline server/src/controllers/eventController.ts:180]: Remove trailing `/`

File: server/src/controllers/eventController.ts [hunk 3]
```diff
@@ -333,112 +305,692 @@ eventController.get('/:id', async (req: Request, res: Response) => {
...
+      return;
+    }
+
+    res.status(500).json({error: 'Internal Server Error'});
+  }
+});
+
+/**
+ * @swagger
+ * /2/events/:
```
Comment (henrykaus) [inline server/src/controllers/eventController.ts:718]: Remove trailing `/`

File: server/src/controllers/eventController.ts [hunk 4]
```diff
@@ -333,112 +305,692 @@ eventController.get('/:id', async (req: Request, res: Response) => {
...
+    if (!eventExists) {
+      res.status(400).json({error: `Event ${id} not found`});
+      return;
+    }
+    res.status(200).json(eventExists);
+    return;
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      res.status(400).json({error: error.message});
+
```
Comment (henrykaus) [inline server/src/controllers/eventController.ts:626]: Remove newlines like this from here down

File: server/src/controllers/eventController.ts [hunk 5]
```diff
@@ -333,112 +305,692 @@ eventController.get('/:id', async (req: Request, res: Response) => {
...
+        seasonid_fk: Number(req.body.seasonid_fk),
+        eventname: req.body.eventname,
+        eventdescription: req.body.eventdescription,
+        active: req.body.active,
+        seasonticketeligible: req.body.seasonticketeligible,
+        imageurl: req.body.imageurl,
+      },
+    });
+    if (!event) {
+      return res.status(400).json({error: `Event ${req.body.eventid} not found `});
```
Comment (henrykaus) [inline server/src/controllers/eventController.ts:765]: Remove trailing space in the string

File: server/src/controllers/eventController.ts [hunk 6]
```diff
@@ -333,112 +305,692 @@ eventController.get('/:id', async (req: Request, res: Response) => {
...
+ *         content:
+ *           application/json:
+ *             schema:
+ *              $ref: '#/components/schemas/Error'
+ *       500:
+ *         description: Internal Server Error. An error occurred while processing the request.
+ */
+eventController.put('/checkin', async (req: Request, res: Response) => {
+  try {
+    const {ticketID, isCheckedIn}= req.body;
```
Comment (henrykaus) [inline server/src/controllers/eventController.ts:978]: Super tiny, but should be a space before the `=`

File: server/src/controllers/eventInstanceController.ts [hunk 7]
```diff
@@ -38,111 +175,111 @@ export const eventInstanceController = Router();
...
  * /2/event-instance:
  *   get:
  *     summary: get all event instances
  *     tags:
- *     - New event instances
+ *     - New event instance
  *     responses:
  *       200:
- *         description: event instance updated successfully.
+ *         description: event instance fetch succesful.
```
Comment (henrykaus) [inline server/src/controllers/eventInstanceController.ts:282]: minor mispeeling

File: server/src/controllers/eventController.ts (hunk 1 above)
Comment (benfaustin95) [reply server/src/controllers/eventController.ts]: Fixed

File: server/src/controllers/eventController.ts (hunk 2 above)
Comment (benfaustin95) [reply server/src/controllers/eventController.ts]: Fixed

File: server/src/controllers/eventController.ts (hunk 4 above)
Comment (benfaustin95) [reply server/src/controllers/eventController.ts]: fixed

File: server/src/controllers/eventController.ts (hunk 5 above)
Comment (benfaustin95) [reply server/src/controllers/eventController.ts]: fixed

File: server/src/controllers/eventController.ts (hunk 6 above)
Comment (benfaustin95) [reply server/src/controllers/eventController.ts]: fixed

File: server/src/controllers/eventInstanceController.ts (hunk 7 above)
Comment (benfaustin95) [reply server/src/controllers/eventInstanceController.ts]: fixed

**Review by henrykaus — APPROVED** [review]

**Pull request merged**